135 adding hrmp channel establishment article (#136)
sora-xor/sora-docs: src/establishing-hrmp-channel.md

# Establish HRMP channel

> To establish communication between your parachain and SORA, you'll need to create HRMP channels that act as bridges for messages. This guide will walk you through the process, using information from the official [Substrate documentation](https://docs.substrate.io/reference/how-to-guides/parachains/add-hrmp-channels/) but applied for SORA.

Reviewer: ```suggestion
To establish communication between your parachain and SORA, you must create HRMP channels that act as bridges for messages. This guide will walk you through the process using information from the official [Substrate documentation](https://docs.substrate.io/reference/how-to-guides/parachains/add-hrmp-channels/) but applied to SORA.
```

Remember that HRMP channels usually go in one direction. For two-way communication, you'll need to set up another channel going the opposite way. Therefore, the following is split in to sub-sections:

## Your Parachain to SORA

1. Obtain the encoded call data of the `hrmp.hrmpInitOpenChannel` extrinsic on the relay chain.

> This special code includes a `recipient` detail that tells relay chain which parachain you would like to connect to. This detail changes depending on what environment you are planning to establish the channel in. For Kusama, look at [this list](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fksm-rpc.stakeworld.io#/parachains), and for Rococo, go to [this list](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frococo-rpc.polkadot.io#/parachains).

Reviewer: ```suggestion
This unique code includes a `recipient` detail telling the relay chain which parachain you want to connect to. This detail changes depending on what environment you are planning to establish the channel in. For Kusama, look at [this list](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fksm-rpc.stakeworld.io#/parachains), and for Rococo, go to [this list](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frococo-rpc.polkadot.io#/parachains).
```

::: info

Combine this parachain id with the other info from the documentation, and you'll end up with something like this for Rococo: `0x3c00db070000e803000000900100`.

:::

> 2. Within your parachain, compose an XCM message that notifies the relay chain about the intent to open a channel with SORA. Refer to the official documentation for an illustrative example of the XCM message.

Reviewer: Please add a link to the docs being referred to here

Author: Added in https://github.com/sora-xor/sora-docs/pull/136/commits/b24258351f26b32b522d6f87da90a59cf7a45743

::: info

> It's important to acknowledge that if your implementation of HRMP channel logic differs from the standard `polkadotXcm.send` pallet, you should modify this step accordingly.

Reviewer: ```suggestion
It is important to acknowledge that if your implementation of HRMP channel logic differs from the standard `polkadotXcm.send` pallet; you should modify this step accordingly.
```

:::

Once the extrinsic executes successfully, SORA needs to accept the channel establishment.

3. Subsequently, the SORA team will accept the channel by incorporating the encoded call data of `hrmp.hrmpAcceptOpenChannel`. This data will feature your parachain's ID as a parameter and will be executed via the `xcmApp.sudoSendXCM` call.

> Upon successful execution, the HRMP establishment from your parachain to SORA is finalized.

Reviewer: ```suggestion
Upon successful execution, (an) HRMP channel(s) will be established from your parachain to SORA.
```
